Emit valid SLD when mm are used as symbol unit

aaime opened the pull request:
SLD does not have a notion of on screen mm, rescale them to pixels to get an equivalent, more valid and more widely usable, output

aaime commented:
Btw, still incomplete, but I wanted to check if its builds because on my machine a bunch of compositing tests are failing all the time (like, even from a un-modified source tree)

m-kuhn commented:
For the failing tests:

Add a pre-commit hook:
https://travis-ci.org/qgis/QGIS/jobs/153943438#L1109-L1114

And make sure your python code is python 3 compatible (no `print 'XYZ'` but `print('XYZ')`)

aaime commented:
Hi m-kuhn, I actually formatted with autopip8 that file, it made a lot of changes, but did not fix that particular error, don't know why? I'm on Linux Mint 18, with the default python packages. I'll fix the print statements

m-kuhn commented:
Hmmm... I always use the pre-commit hook which I find intriguingly easy to use.

aaime committed:
Emit valid SLD when on screen mm are used

SLD does not have a notion of on screen mm, rescale them to pixels to get an equivalent, more valid and more widely usable, output

aaime commented:
Travis build passing :-)

m-kuhn commented:
Looks good to me
You will have to put the new test into the CMakeLists.txt file so they are run with ctest (and therefore travis)

aaime commented:
Hi @m-kuhn  I actually extended a test file I added in a previous commit, it should already be in CMakeLists.txt. Checking... yep:

````
        Start 223: PyQgsSymbolLayerCreateSld
223/251 Test #223: PyQgsSymbolLayerCreateSld ...............   Passed    0.40 sec
`````
